components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -6738,4 +6812,122 @@ private void rollbackAddApplicationTransaction(Connection connection, ServicePro
             throw new IdentityApplicationManagementException("Error while rolling back the transaction.", e);
         }
     }
+
+    /**
+     * Retrieve the discoverable groups for the application.
+     *
+     * @param applicationId Application ID.
+     * @param connection    Database connection.
+     * @param tenantDomain  Tenant domain of the application.
+     * @return Discoverable groups list.
+     * @throws IdentityApplicationManagementException If an error occurred while retrieving discoverable groups.
+     */
+    private DiscoverableGroup[] getDiscoverableGroups(int applicationId, Connection connection, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Retrieving discoverable groups for application with ID: " + applicationId);
+        }
+
+        Map<String, List<GroupBasicInfo>> groupInfoMap = new HashMap<>();
+        try (PreparedStatement statement = connection.prepareStatement(
+                ApplicationMgtDBQueries.GET_GROUP_ASSOCIATIONS_BY_APP_ID)) {
+            statement.setInt(1, applicationId);
+            try (ResultSet resultSet = statement.executeQuery()) {
+                AbstractUserStoreManager userStoreManager =
```

Comment:
try to refactor if possible

Reply:
https://github.com/wso2/carbon-identity-framework/pull/6283/commits/25cd6027c99764766e277a7a1d9987e830cc3b4e